third_party/move/move-compiler/src/parser/syntax.rs

```diff
@@ -2766,9 +2832,14 @@ fn parse_struct_variant(context: &mut Context) -> Result<(StructVariant, bool),
                 "a field",
             )?,
             true,
+            false,
         )
+    } else if context.tokens.peek() == Tok::LParen {
+        let loc = current_token_loc(context.tokens);
+        require_move_2(context, loc, "positional fields");
+        (parse_anonymous_fields(context)?, true, true)
```

Comment:
Could you explain why `has_block` is true here? Would we consider `()` to be the same as `{}` when it comes to expecting a comma between variants?

We also need some tests for this case.

Reply:
It's a bug. Fixed now. Testes in `third_party/move/move-compiler-v2/tests/checking/positional_fields/variant_decl_invalid.move`